pkg/clients/tsclient/client.go

```diff
@@ -443,7 +444,10 @@ func (c *HTTPClient) Flush() error {
 		}
 		return err
 	}
-	defer c.handleSonarResponse(resp.Body)
+	contentType := resp.Header.Get(contentTypeHeader)
+	if contentType == jsonContentType {
+		defer c.handleSonarResponse(resp.Body)
```

Comment:
Any idea why this was deferred in the first place?

Reply:
I'm not positive tbh